.github/workflows/e2e_targeted.yaml

```diff
@@ -0,0 +1,78 @@
+name: e2e tests (targeted)
+
+on:
+  schedule:
+    - cron: '30 4 * * *'
+  workflow_dispatch:
+    inputs:
+      kubernetes-version:
+        description: 'Kubernetes version to test with'
+        required: true
+        default: 'v1.23.1'
+      istio-version:
+        description: 'Istio version to test with'
+        required: true
+        default: 'v1.12.2'
+      controller-image:
+        description: 'KIC Docker image to test with. The default "localhost:5000/kong/kubernetes-ingress-controller:ci" builds an image from the dispatch branch'
+        required: true
+        default: 'localhost:5000/kong/kubernetes-ingress-controller:ci'
+
+jobs:
+  e2e-tests:
+    environment: "Configure ci"
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+    - name: setup golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: '^1.17'
+
+    - name: cache go modules
+      uses: actions/cache@v2.1.7
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-build-codegen-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-build-codegen-
+
+    - name: checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Set up QEMU
+      if: ${{ github.event.inputs.controller-image == 'localhost:5000/kong/kubernetes-ingress-controller:ci' }}
+      uses: docker/setup-qemu-action@v1
```

Comment:
So the custom build requires qemu eh? :thinking:

Reply:
Possibly? It was in the example at https://github.com/docker/build-push-action/blob/master/docs/advanced/local-registry.md, but that doesn't provide any context as to why it's there.